Kubernetes 1.12

aermakov-zalando opened the pull request:
* Kubernetes: update to 1.12.5.
 * CA: update to v1.12.1-internal10
 * Enable `TTLAfterFinished` feature flag

aermakov-zalando committed:
Update Kubernetes to v1.12.4

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Support multiple instance types for worker pools

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs commented on cluster/node-pools/master-default/userdata.clc.yaml:
```diff
@@ -167,7 +167,6 @@ systemd:
       --cni-conf-dir=/etc/kubernetes/cni/net.d \
       --network-plugin=cni \
       --container-runtime=docker \
-      --cadvisor-port=4194 \
```
on purpose?

mikkeloscar replied:
Yes, this flag is removed in 1.12 and replaced by the cadvisor-daemonset in our setup.

szuecs commented on cluster/node-pools/worker-default/userdata.clc.yaml:
```diff
@@ -159,7 +159,6 @@ systemd:
       --cni-conf-dir=/etc/kubernetes/cni/net.d \
       --network-plugin=cni \
       --container-runtime=docker \
-      --cadvisor-port=4194 \
```
on purpose?

aermakov-zalando committed:
Determine whether an instance is a Spot one dynamically

With MixedInstancesPolicy ASGs, we cannot rely on LaunchTemplate to
determine whether an instance is an On Demand one or not. Determine
whether we're running on Spot on startup and configure stuff accordingly
instead of hardcoding it into the user data.

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Enable TTLAfterFinished (apiserver, controller-manager)

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1670 from zalando-incubator/dev-to-kube-1.12

Dev to kube 1.12

aermakov-zalando committed:
Merge pull request #1668 from zalando-incubator/enable-ttl-after-finished

Enable TTLAfterFinished (apiserver, controller-manager)

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1676 from zalando-incubator/dev-to-kube-1.12

Dev to kube 1.12

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

mikkeloscar committed:
Merge pull request #1677 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

aermakov-zalando committed:
Add a feature flag for the custom CA build

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1666 from zalando-incubator/multi-asg

Support multiple instance types for worker pools

aermakov-zalando committed:
Fix the template for worker-default node pool

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1681 from zalando-incubator/fix-pool

Fix worker-default node pool

aermakov-zalando committed:
Fix worker-default pool again

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

mikkeloscar committed:
Merge pull request #1682 from zalando-incubator/fix-pool-again

Fix worker-default pool again

mikkeloscar committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

mikkeloscar committed:
Merge pull request #1687 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

mikkeloscar committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

gargravarr committed:
Merge pull request #1689 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.12

aermakov-zalando committed:
Merge pull request #1700 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

szuecs committed:
enable zmon from root account to query AWS metrics in other accounts

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
cleanup deletion

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
Merge pull request #1704 from zalando-incubator/hotfix/enable-zmon-to-query-cloudwatch

Hotfix/enable zmon to query cloudwatch

aermakov-zalando committed:
Always set the CLC labels

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
Merge pull request #1709 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

aermakov-zalando committed:
Merge pull request #1708 from zalando-incubator/clc-labels

Always set the CLC labels

aermakov-zalando committed:
CA: use expander=prefer-spot

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs committed:
Merge pull request #1712 from zalando-incubator/dev-to-kube-1.12

Dev to kube 1.12

szuecs committed:
Merge pull request #1711 from zalando-incubator/experimental-autoscaler-bump

CA: use expander=prefer-spot

aermakov-zalando committed:
Drop reflector metrics from Kubelet

For some reason these metrics contain ~100-150 series for every single
resource in the cluster. Let's just drop them completely.

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs committed:
Merge pull request #1716 from zalando-incubator/kubelet-disable-reflector-metrics

Drop reflector metrics from Kubelet

mikkeloscar committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

mikkeloscar committed:
Merge pull request #1717 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

mikkeloscar committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

mikkeloscar committed:
Merge pull request #1720 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

aermakov-zalando committed:
CoreDNS: add dnsmasq for caching

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

mikkeloscar committed:
Merge pull request #1728 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

mikkeloscar committed:
Update to Kubernetes v1.12.5

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.12.md#v1125

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

szuecs committed:
Merge pull request #1741 from zalando-incubator/increase-elb-timeout

API server: increase the ELB timeout

arjunrn committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Arjun Naik <arjun.rn@gmail.com>

arjunrn committed:
Merge pull request #1744 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

aermakov-zalando committed:
Update CA

 * Update to v1.12.2
 * MixedInstancesPolicy support + prefer-spot expander

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1742 from zalando-incubator/kube-1.12-5

Update to Kubernetes v1.12.5

mikkeloscar committed:
Merge pull request #1745 from zalando-incubator/autoscaler-1.12.2

Update CA

arjunrn committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Arjun Naik <arjun.rn@gmail.com>

arjunrn committed:
Merge pull request #1748 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

arjunrn committed:
Update to newer version of kube-metrics-adapter

Signed-off-by: Arjun Naik <arjun.rn@gmail.com>

arjunrn committed:
Merge pull request #1752 from zalando-incubator/missing-backend

Update to newer version of kube-metrics-adapter

aermakov-zalando committed:
Add category: all to CRDs

 * Show PCS, ingress templates, VPAs by default in `get all`
 * Fix the hack causing PCS definitions to be duplicated

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

mikkeloscar committed:
Remove rbac feature flag

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

szuecs committed:
Merge pull request #1757 from zalando-incubator/category-all

Add category: all to CRDs

aermakov-zalando committed:
Fix ports in other places as well

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
CLC: tolerate not-ready taint

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Use the correct feature flag in default-limits

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Update admission-controller

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Disable admission controller for statefulset e2e namespace

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

mikkeloscar committed:
Merge pull request #1765 from zalando-incubator/fix-flaky-sstests

Disable admission controller for statefulset e2e namespace

aermakov-zalando committed:
Admission controller: handle node admissions

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1763 from zalando-incubator/clc-tolerate-not-ready

Preparation steps for TaintNodesByCondition

szuecs committed:
Merge pull request #1733 from zalando-incubator/coredns-with-cache

CoreDNS: add dnsmasq for caching

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.12

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

aermakov-zalando committed:
dnsmasq: fix configuration to ignore resolv.conf

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs committed:
Merge pull request #1770 from zalando-incubator/fix-dnsmasq

dnsmasq: fix configuration to ignore resolv.conf

szuecs committed:
Merge pull request #1767 from zalando-incubator/dev-to-kube-1.12

dev-to-kube-1.12

aermakov-zalando commented:
👍

jrake-revelant commented:
👍
